cpp/bench/ann/src/common/benchmark.hpp

```diff
@@ -506,7 +506,9 @@ inline auto run_main(int argc, char** argv) -> int
     return -1;
   }
 
+#ifndef CPU_ONLY
   if (!CUDART_FOUND) { log_warn("cudart library is not found, GPU-based indices won't work."); }
+#endif
```

Comment:
I suggest updating the `CUDART_FOUND` definition at the top of the file instead of nesting the pragmas here.